frontend/app/src/components/StreamlitDialog/DeployDialog/DeployDialog.tsx

```diff
@@ -35,16 +35,36 @@ import Rocket from "@streamlit/app/src/assets/svg/rocket.svg"
 import {
   STREAMLIT_COMMUNITY_CLOUD_DOCS_URL,
   STREAMLIT_DEPLOY_TUTORIAL_URL,
+  DEPLOY_URL,
+  STREAMLIT_CLOUD_URL,
 } from "@streamlit/app/src/urls"
 import {
   DetachedHead,
   ModuleIsNotAdded,
   NoRepositoryDetected,
 } from "@streamlit/app/src/components/StreamlitDialog/DeployErrorDialogs"
-import { getDeployAppUrl } from "@streamlit/app/src/components/MainMenu/MainMenu"
 
 const { GitStates } = GitInfo
 
+export const getDeployAppUrl = (gitInfo: IGitInfo | null): (() => void) => {
+  // If not in git repo, direct them to the Streamlit Cloud page.
+  let deployUrlStr = STREAMLIT_CLOUD_URL
+  if (gitInfo) {
+    // If the app was run inside a GitHub repo, autofill for a one-click deploy.
+    // E.g.: https://share.streamlit.io/deploy?repository=melon&branch=develop&mainModule=streamlit_app.py
+    const deployUrl = new URL(DEPLOY_URL)
+
+    deployUrl.searchParams.set("repository", gitInfo.repository || "")
+    deployUrl.searchParams.set("branch", gitInfo.branch || "")
+    deployUrl.searchParams.set("mainModule", gitInfo.module || "")
+    deployUrlStr = deployUrl.toString()
+  }
+
+  return (): void => {
+    window.open(deployUrlStr, "_blank")
```

Comment:
Do we want to use getWindowInOpenCallback that u exported?

Reply:
I initially used `getWindowInOpenCallback`, but I thought it was not worth it to create a dependency between the dialog and main menu for a one-liner.

Reply:
Hm I'll leave it up to you. I think we should use it so that if we change it in one method, all of the other ones get changed more easily but I can also see if this component only needs the change, it doesn't make a difference.

Reply:
I did another iteration on this. I don't think we need to use this strange callback pattern here, and also there are two other `window.open` in the DeployDialog as well. So, did a small refactoring to unify this within the component.